src/routes/validationRoute.js

```diff
@@ -37,12 +38,18 @@ router.get('/login', (req, res) => {
 
 router.post('/login', async (req, res) => {
     if (!req.body.username || !req.body.password) return res.status(400).send({ error: 'bad request' })
-    const user = await User.findOne({ username: req.body.username })
-    const isMatch = await bcrypt.compare(req.body.password, user.password)
+    const username = req.body.username.trim()
```

Comment:
Ensure that user inputs are properly validated to prevent injection attacks. Try using a library like zod

Reply:
https://www.turing.com/blog/data-integrity-through-zod-validation/